Worker/src/Unified/GridWorker.cs

```diff
@@ -223,16 +258,89 @@ public byte[] Execute(ITaskHandler taskHandler)
     return null;
   }
 
+
   public void SessionFinalize()
-  {
-  }
+    => OnSessionLeave();
 
   public void DestroyService()
-    => Dispose();
+  {
+    OnDestroyService();
+    Dispose();
+  }
 
   /// <summary>
   ///   Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources
   /// </summary>
   public void Dispose()
     => SessionFinalize();
+
+  public void OnCreateService()
+  {
+    if (ServiceClass is ISessionConfiguration iSessionConfiguration)
+    {
+      using (AssemblyLoadContext.EnterContextualReflection(ServiceClass.GetType()
+                                                                       .Assembly))
+      {
+        iSessionConfiguration.OnCreateService(serviceContext_);
+      }
+    }
+  }
+
+
+  /// <summary>
+  ///   The internal function onSessionEnter to openSession for clientService under GridWorker
+  /// </summary>
+  /// <param name="session"></param>
+  public void OnSessionEnter(Session session)
+  {
+    sessionContext_ = new SessionContext
+                      {
+                        ClientLibVersion = GridAppVersion,
+                        SessionId        = session.Id,
+                      };
+    SessionId = session;
+
+
+    if (ServiceClass is ISessionConfiguration iSessionConfiguration)
+    {
+      if (iSessionConfiguration.SessionId == null || string.IsNullOrEmpty(iSessionConfiguration.SessionId.Id))
+      {
+        iSessionConfiguration.SessionId = session;
+      }
+
+      iSessionConfiguration.SessionId = session;
```

Comment:
Either the `if` is useless, or the assignment after the `if` should disappear.

Reply:
indeed. updated